internal/js_library/js_library.bzl

```diff
@@ -223,7 +227,9 @@ def _impl(ctx):
     ]
 
     if ctx.attr.package_name:
-        path = "/".join([p for p in [ctx.bin_dir.path, ctx.label.workspace_root, ctx.label.package] if p])
+        path = "/".join([
+            p for p in [ctx.bin_dir.path, ctx.label.workspace_root, ctx.label.package, ctx.attr.strip_path] if p
```

Comment:
do you think we can do any decent error messaging if you have a typo or misunderstanding, and the strip_prefix doesn't match against one or more of the srcs?

Reply:
I have added a check that at least one source is within the stripped path. However, an alternative would be to fail if any src is not in the stripped path. Any preference?